### PR TITLE
Extended DRAT signature to operational DRAT

### PR DESCRIPTION
**Pull request by alex-ozdemir**
The DRAT signature now supports both operational and specified DRAT.
That is, either kind of proof will be accepted.

The goal of this implementation of operational DRAT was to re-use as
much of the specified DRAT machinery as possible. However, by writing a
separate operational signature, we could make it much more efficient
(after all, operational DRAT came about because of a push for efficient
cheking).

You can run the new AND old DRAT tests by running

```
lfscc sat.plf smt.plf lrat.plf drat.plf drat_test.plf
```

### CODE REVIEW

File: proofs/signatures/drat.plf [hunk 1]
```diff
@@ -229,21 +229,158 @@
...
+
+; Remove the partial assignment represented by c from the global assignment
+(program clause_undo_into_global ((a clause)) Unit
+         (match a
+                (cln unit)
+                ((clc l rest)
+                 (do (lit_un_mk_sat l) (clause_undo_into_global rest)))))
+
+; Does this clause have no floating literals w.r.t. the global assignment?
+(program clause_no_floating ((c clause)) bool
```
Comment (yoni206) [inline proofs/signatures/drat.plf]: Are the literals floating w.r.t. the global assignment itself? or just its domain? Are the (globally) assigned values relevant?

File: proofs/signatures/drat.plf [hunk 2]
```diff
@@ -229,21 +229,158 @@
...
+(program clause_one_sat ((c clause)) bool
+         (match c
+                (cln ff)
+                ((clc l rest) (match (lit_is_sat l)
+                                    (tt (clause_no_sat rest))
+                                    (ff (clause_one_sat rest))))))
+
+; Is this clause a unit clause w.r.t. the global assignment?
+; This notion is defined as:
+;    * A clause where no literals are floating, and one is sat.
```
Comment (yoni206) [inline proofs/signatures/drat.plf]:
```suggestion
;    * A clause where no literals are floating, and exactly one is sat.
```

File: proofs/signatures/drat.plf [hunk 3]
```diff
@@ -229,21 +229,158 @@
...
+                                      (is_operational_drat_proof_h f' up_model' pf'))))
+                          (ff ff))))))
+
+; Is this an operational DRAT proof of bottom for this formula?
+(program is_operational_drat_proof ((f cnf) (pf DRATProof)) bool
+         (match (build_up_model f)
+                (UPCRBottom tt)
+                ((UPCRModel model) (is_operational_drat_proof_h f model pf))))
+
+; Is this a specified or operation DRAT proof of bottom for this formula?
```
Comment (yoni206) [inline proofs/signatures/drat.plf]:
```suggestion
; Is this a specified or operational DRAT proof of bottom for this formula?
```

**Comment by 4tXJ7f** [comment]
Marking PR as "do not merge" to give @alex-ozdemir the chance to consider @yoni206's suggestions.

**Pull request merged**